nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/TrashAction.java

```diff
@@ -119,6 +122,12 @@ public void setSystemProperty(CoreSession session, List<String> ids, boolean val
                     DocumentRef ref = new IdRef((String) map.get(ECM_UUID));
                     try {
                         session.setDocumentSystemProp(ref, "isTrashed", value);
+                        String docName = (String) map.get(ECM_NAME);
+                        if (!value && Framework.getService(TrashService.class).isMangledName(docName)) {
```

Comment:
Get the service before the loop so it is not retrieve for each document

Reply:
I did it this way so it's not retrieved at all if the first check `!value` is false
Otherwise I can still put it in an external local variable as you propose.
Wdyt ?

Reply:
Anyway, you have duplicated `Framework.getService(TrashService.class)`, so better to just create a variable for it.